packages/polaris-viz/src/components/Grid/components/GroupCell.tsx

```diff
@@ -0,0 +1,144 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import type {ScaleLinear} from 'd3-scale';
+
+import styles from '../Grid.scss';
+import type {CellGroup} from '../Grid';
+
+import {Background} from './Background';
+import {GroupInfo} from './GroupInfo';
+
+interface GroupCellProps {
+  group: CellGroup;
+  xScale: ScaleLinear<number, number>;
+  cellHeight: number;
+  cellWidth: number;
+  isSmallContainer: boolean;
+  hoveredGroups: Set<string>;
+  handleGroupHover: (
+    group: CellGroup | null,
+    event: React.MouseEvent | React.KeyboardEvent | React.FocusEvent,
+  ) => void;
+  getColors: (group: CellGroup) => {bgColor: string; textColor: string};
+  containerWidth: number;
+  isAnimated: boolean;
+  index: number;
+}
+
+export const GroupCell: React.FC<GroupCellProps> = ({
+  group,
+  xScale,
+  cellHeight,
+  cellWidth,
+  isSmallContainer,
+  hoveredGroups,
+  handleGroupHover,
+  getColors,
+  containerWidth,
+  isAnimated,
+  index,
+}) => {
+  const [prefersReducedMotion, setPrefersReducedMotion] = useState(false);
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia('(prefers-reduced-motion: reduce)');
+    setPrefersReducedMotion(mediaQuery.matches);
+
+    const handleChange = () => setPrefersReducedMotion(mediaQuery.matches);
+    mediaQuery.addEventListener('change', handleChange);
+
+    return () => mediaQuery.removeEventListener('change', handleChange);
+  }, []);
+
+  const groupWidth = (group.end.col - group.start.col + 1) * cellWidth;
+  const groupHeight = (group.end.row - group.start.row + 1) * cellHeight;
+  const groupValue = group.value;
+  const groupSecondaryValue = group.secondaryValue;
+
+  const isHovered = hoveredGroups.size === 0 || hoveredGroups.has(group.name);
+  const opacity = isHovered ? 1 : 0.3;
+
+  const groupNameOffset = 10;
+  const showNameAndSecondaryValue = containerWidth > 500;
```

Comment:
I think we want to keep this one slightly bigger, otherwise there is no enough space from 400px to 500px to show both the value and secondary value. I move this into a constant explaining the meaning of 500 tho.